### PR TITLE
add callback uri params

### PR DESCRIPTION
**Pull request by meotimdihia**
We can revert this change after simple-oauth2's issue is implemented:  https://github.com/lelylan/simple-oauth2/pull/208

### CODE REVIEW

**Comment by mcollina** [comment]
Can you add a unit test for this change? So we don’t regress.

**Comment by meotimdihia** [comment]
Hi, I add a unit test.

File: index.js [hunk 1]
```diff
@@ -45,19 +48,21 @@ const oauthPlugin = fp(function (fastify, options, next) {
...
   const scope = options.scope
   const generateStateFunction = options.generateStateFunction || defaultGenerateStateFunction
   const checkStateFunction = options.checkStateFunction || defaultCheckStateFunction
   const startRedirectPath = options.startRedirectPath
 
   function startRedirectHandler (request, reply) {
     const state = generateStateFunction()
-
-    const authorizationUri = this[name].authorizationCode.authorizeURL({
+    const urlOptions = Object.assign(callbackUriParams, {
```
Comment (allevo) [inline index.js]: This changes the original object. Could you change this line in order to avoid the initial object modification?

Comment (meotimdihia) [reply index.js]: Hi, I fixed it 👍 .

File: test.js [hunk 2]
```diff
@@ -210,6 +210,28 @@ t.test('options.callbackUri should be and object', t => {
...
+    name: 'the-name',
+    credentials: {
+      client: {
+        id: 'my-client-id',
+        secret: 'my-secret'
+      },
+      auth: oauthPlugin.GITHUB_CONFIGURATION
+    },
+    callbackUri: '/callback',
+    callbackUriParams: 1
```
Comment (allevo) [inline test.js]: There is no test for this value. Could you invoke `startRedirectHandler` in order to check the redirection has this value in the query string?

Comment (meotimdihia) [reply test.js]: Hi, please help me check if i did right 💯 .

Comment (Atinux) [reply test.js]: Friendly ping to @allevo since @meotimdihia pushed another commit including the test

**Comment by ericmacfa** [comment]
Based on the discussion in [lelylan/simple-oauth2#208](https://github.com/lelylan/simple-oauth2/pull/208#issuecomment-421600880), this PR is essentially the recommended approach for defining additional parameters on the authorization URL 😄

This PR would be very helpful, thanks for doing this work!

File: index.js [hunk 3]
```diff
@@ -29,6 +29,9 @@ const oauthPlugin = fp(function (fastify, options, next) {
   if (typeof options.callbackUri !== 'string') {
     return next(new Error('options.callbackUri should be a string'))
   }
+  if (options.callbackUriParams && typeof options.callbackUriParams !== 'object') {
+    return next(new Error('options.generateStateFunction should be a object'))
```
Comment (Eomm) [inline index.js]:
```suggestion
    return next(new Error('options.callbackUriParams should be a object'))
```

**Pull request merged**